### DISCUSSION TITLE
Github Pages fails to load a subdirectory / xhr load error

### DISCUSSION THREAD

**Discussion by y0nei** [community/community#38191] [Other Feature Feedback, Questions, & Ideas]
Im trying to deploy a simple project [extensible-js-taggable-gallery](https://github.com/ziimro/extensible-js-taggable-gallery), i use xhr to load a directory but gh pages fails to initalize it. Upon navigating to `https://ziimro.github.io/extensible-js-taggable-gallery/galleryimages` it just returns 404.
I have tried it with the default jekyll config and base github pages deployment and both failed.

And i have tried different values for my baseUrl in my `plain.js` file: `"galleryimages"`, `"./galleryimages/"`, `"galleryimages/"`, but all have failed
```js
var baseUrl = "./galleryimages";
```

**Comment by queenofcorgis** [comment]
Hi @Ziimro,

The GitHub Community on Discussions is a space for all GitHub users to come and discuss ideas and projects. Your additional post was removed because it contained content that violates our Community Code of Conduct, including posting the same comment in multiple places. As a result, we have hidden the duplicate comments, and we ask that you no longer cross-post.

For additional guidance, please review our [Community on Discussions Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md) and [GitHub’s Terms of Service](https://docs.github.com/en/github/site-policy/github-terms-of-service).

**Reply by yoannchaudet** [reply]
GitHub Pages does not enumerate content of arbitrary folders.

https://ziimro.github.io/extensible-js-taggable-gallery/galleryimages will return you a 404 because you don't have an `index.html` file to serve from that folder.

Your images are there and being served properly if you access them: https://ziimro.github.io/extensible-js-taggable-gallery/galleryimages/dog%201%20pet.jpg

**Reply by y0nei** [reply]
> GitHub Pages does not enumerate content of arbitrary folders.
> 
> https://ziimro.github.io/extensible-js-taggable-gallery/galleryimages will return you a 404 because you don't have an `index.html` file to serve from that folder.
> 
> Your images are there and being served properly if you access them: https://ziimro.github.io/extensible-js-taggable-gallery/galleryimages/dog%201%20pet.jpg

How would i allow jekyll to serve a directory? In nginx i have worked around the issue enabling `autoindex on;` on the galleryimages folder

**Reply by yoannchaudet** [reply]
For Jekyll you should enumerate folders at build time only. E.g. https://stackoverflow.com/a/45221830
